hyperspy/_signals/signal1d.py

```diff
@@ -1033,20 +1033,27 @@ def _remove_background_cli(
             signal_range[0],
             signal_range[1],
             only_current=False)
-        if not fast:
+        if fast:
```

Comment:
The multi-dimensional approach is actually specially well suited for lazy computation. This would bring the extra advantages of parallelization and avoid the memory issue. So, I wonder if it wouldn't be possible to implement this using dask as follows (pseudocode):

```python
if lazy:
    # Get map/x chunks from the signal navigation/signal chunks
    # Create lazy parameters and ``x`` arrays (hence  ``nd_function`` output) using these chunks.
else:
   # Create lazy parameters and ``x`` arrays (hence  ``nd_function`` output) using dask's new auto chunking.
   # Call ``compute`` at the end.
```
The lazy computation could be controlled via a kwarg with default to True e.g. ``remove_background(parallel=True)``.

Reply:
Actually, this should in the `as_signal` method of the model and check if the `function_nd` is available for all components and use this approach if necessary.

Reply:
I fully agree.

Reply:
Currently ``as_signal`` operates in parallel but pixel by pixel, what should be much slower than performing the same operation using dask vectorially. This is very interesting because it should be possible to write all but the ``EELSCLEdge`` component using ``Expression``.

Reply:
Yes, we needs to do this.
I had a look at this and I could not get it to work in the time given. I will not have time to do this properly now, so I will open an issue to keep it for later. In the mean time, since this is an improvement of the current implementation, I would suggest to keep this approach for now.